include/roboteam_ai/world/FieldMessage.h

```diff
@@ -14,124 +10,234 @@
+    std::map<std::string, std::string> NAME_MAP = {
+        std::make_pair("TopTouchLine", "top_line"),
+        std::make_pair("BottomTouchLine", "bottom_line"),
+        std::make_pair("LeftGoalLine", "left_line"),
+        std::make_pair("RightGoalLine", "right_line"),
+        std::make_pair("HalfwayLine", "half_line"),
+        std::make_pair("CenterLine", "center_line"),
+        std::make_pair("LeftPenaltyStretch", "left_penalty_line"),
+        std::make_pair("RightPenaltyStretch", "right_penalty_line"),
+        std::make_pair("LeftFieldLeftPenaltyArc", "top_left_penalty_arc"),
+        std::make_pair("LeftFieldRightPenaltyArc", "bottom_left_penalty_arc"),
+        std::make_pair("RightFieldLeftPenaltyArc", "top_right_penalty_arc"),
+        std::make_pair("RightFieldRightPenaltyArc", "bottom_right_penalty_arc"),
+        std::make_pair("LeftFieldLeftPenaltyStretch", "top_left_penalty_stretch"),
+        std::make_pair("LeftFieldRightPenaltyStretch", "bottom_left_penalty_stretch"),
+        std::make_pair("RightFieldLeftPenaltyStretch", "bottom_right_penalty_stretch"),
+        std::make_pair("RightFieldRightPenaltyStretch", "top_right_penalty_stretch"),
+        std::make_pair("CenterCircle", "center_circle"),
+    };
+
+    // Used to convert field line name, in string format, to the corresponding FieldLineName enum value
+    std::map<std::string, FieldLineName> CONVERT_TO_FIELD_LINE_NAME = {
+        std::make_pair("top_line", TOP_LINE),
+        std::make_pair("bottom_line", BOTTOM_LINE),
+        std::make_pair("left_line", LEFT_LINE),
+        std::make_pair("right_line", RIGHT_LINE),
+        std::make_pair("half_line", HALF_LINE),
+        std::make_pair("center_line", CENTER_LINE),
+        std::make_pair("left_penalty_line", LEFT_PENALTY_LINE),
+        std::make_pair("right_penalty_line", RIGHT_PENALTY_LINE),
+        std::make_pair("top_left_penalty_stretch", TOP_LEFT_PENALTY_STRETCH),
+        std::make_pair("bottom_left_penalty_stretch", BOTTOM_LEFT_PENALTY_STRETCH),
+        std::make_pair("top_right_penalty_stretch", TOP_RIGHT_PENALTY_STRETCH),
+        std::make_pair("bottom_right_penalty_stretch", BOTTOM_RIGHT_PENALTY_STRETCH)
+    };
+
+    // Used to convert field arc name, in string format, to the corresponding FieldArcName enum value
+    std::map<std::string, FieldArcName> CONVERT_TO_FIELD_ARC_NAME = {
+        std::make_pair("center_circle", CENTER_CIRCLE)
+    };
+
+private:
+    std::map<FieldValueName, double> fieldValues = {}; // Stores all the constant of the field (lengths, widths, positions)
+    std::map<FieldLineName, FieldLineSegment> fieldLines = {}; // Stores all the lines of the field
+    std::map<FieldArcName, FieldArc> fieldArcs = {}; // Stores all the arcs of the field
+    std::map<FieldVectorName, Vector2> fieldVectors = {}; // Stores all positions of the field
+
+public:
+    /**
+     * Constructor that creates an unitialized FieldMessage
+     */
+    FieldMessage() = default;
```

Comment:
That will require some more work to do and is indeed important to do. However Jesse and I agreed that the behavior tree needs more attention. So this will be left to do for in the future.